Sources_v3/StreamChat/Controllers/CurrentUserController/CurrentUserController_Tests.swift

```diff
@@ -565,6 +565,182 @@ final class CurrentUserController_Tests: StressTestCase {
+        
+        // Call update user
+        var completionCalled = false
+        controller.updateUserData(
+            name: expectedName,
+            imageURL: expectedImageUrl,
+            userExtraData: nil,
+            completion: { _ in
+                completionCalled = true
+            }
+        )
+        
+        // Simulate API response
+        let currentUserUpdateResponse = UserUpdateResponse(
+            user: UserPayload.dummy(
+                userId: userPayload.id,
+                name: expectedName,
+                imageUrl: expectedImageUrl
+            )
+        )
+        client.mockAPIClient.test_simulateResponse(.success(currentUserUpdateResponse))
+        
+        var currentUser: CurrentChatUser? {
+            client.databaseContainer.viewContext.currentUser()?.asModel()
+        }
+        
+        // Check the completion is called and the current user model was updated
+        AssertAsync {
+            Assert.willBeTrue(completionCalled)
+            Assert.willBeEqual(currentUser?.name, expectedName)
+            Assert.willBeEqual(currentUser?.imageURL, expectedImageUrl)
+        }
+    }
+    
+    func test_updateUser_whenAPICallError_shouldCompleteWithError() throws {
+        // Simulate user already set
+        let userPayload: CurrentUserPayload<DefaultExtraData.User> = .dummy(userId: .unique, role: .user)
+        try client.databaseContainer.writeSynchronously {
+            try $0.saveCurrentUser(payload: userPayload)
+        }
+        
+        // Call update user
+        var completionError: Error?
+        controller.updateUserData(
+            name: .unique,
+            imageURL: nil,
+            userExtraData: nil,
+            completion: { error in
+                completionError = error
+            }
+        )
+        
+        // Simulate API error
+        let error = TestError()
+        client
+            .mockAPIClient
+            .test_simulateResponse(
+                Result<UserUpdateResponse<DefaultExtraData.User>, Error>.failure(error)
+            )
+        client
+            .mockAPIClient
+            .cleanUp()
+        
+        // Assert the completion is called with the error
+        AssertAsync.willBeEqual(completionError as? TestError, error)
+    }
+    
+    func test_updateUser_whenCurrentUserDoesNotExist_shouldError() throws {
+        let error = try await {
+            controller.updateUserData(
+                name: .unique,
+                imageURL: nil,
+                userExtraData: nil,
+                completion: $0
+            )
+        }
+        
+        XCTAssert(error is ClientError.CurrentUserDoesNotExist)
+    }
+    
+    func test_updateUser_whenNoDataProvided_shouldNotMakeRequest() throws {
+        _ = try await {
+            controller.updateUserData(
+                name: nil,
+                imageURL: nil,
+                userExtraData: nil,
+                completion: $0
+            )
+        }
```

Comment:
How come this test passes? You never call completion so the `await` should time-out 🤔

Reply:
The test was actually bad. I had to first simulate that a user was already logged in to properly test this. So in fact, the await did not timeout because the completion was actually being called, but in the first error case, and not this one.